kubernetes/schema_webhook_client_config.go

```diff
@@ -53,6 +56,32 @@ func webhookClientConfigFields() map[string]*schema.Schema {
 			Type:        schema.TypeString,
 			Description: apiDoc["url"],
 			Optional:    true,
+			ValidateFunc: func(v interface{}, k string) ([]string, []error) {
+				u, err := url.Parse(v.(string))
+				if err != nil {
+					return nil, []error{err}
+				}
+
+				errs := []error{}
```

Comment:
Good idea! This will create a more helpful error message. My IDE is being nit-picky and saying that nil slices should be created like this instead `var errs []error`, but I don't mind it either way.

Reply:
I know that in this case the validator checks the length rather than checking if it's `nil` so it is a matter of taste here. 

Interestingly, a nil slice will behave the same as an empty slice in every way except that will also return true on comparison to `nil`. See this Stackoverflow post: https://stackoverflow.com/questions/44305170/nil-slices-vs-non-nil-slices-vs-empty-slices-in-go-language